File: android-utils/src/main/java/com/gurunars/android_utils/ColoredShapeDrawable.java
package com.gurunars.android_utils;

import android.graphics.drawable.ShapeDrawable;
import android.graphics.drawable.shapes.Shape;

/**
 * Drawable with customizable shape and color.
 */
public final class ColoredShapeDrawable extends ShapeDrawable {

    public ColoredShapeDrawable(Shape shape, int color) {
        super(shape);
        getPaint().setColor(color);
    }

}

File: android-utils/src/main/kotlin/com/gurunars/android_utils/ColoredShapeDrawable.kt
package com.gurunars.android_utils

import android.graphics.drawable.ShapeDrawable
import android.graphics.drawable.shapes.Shape

/**
 * Drawable with customizable shape and color.
 */
class ColoredShapeDrawable(shape: Shape, color: Int) : ShapeDrawable(shape) {

    init {
        paint.color = color
    }

}
